File: Todo app/todo api/src/models/todo.js
import Mongoose from 'mongoose'
import Joi from 'joi'

const todoSchema = new Mongoose.Schema({
    userid: {
        type: String,
        required: true,
        trim: true,
        minlength: 1,
        maxlength: 255,
    },
    title: {
        type: String,
        required: true,
        trim: true,
        minlength: 5,
        maxlength: 255,
    },
    completed: {
        type: Boolean,
        default: false
    },
    due: {
        type: String,
        default: '09-11-2022'
    },
    description: {
        type: String,
        default: ''
    }
})


// joi
const TodoModel = Mongoose.model('Todo', todoSchema)

const JoiSchema = Joi.object({
    userid: Joi.string().min(1).max(255).required(),
    title: Joi.string().min(5).max(255).required(),
    completed: Joi.boolean(),
    due: Joi.string(),
    description: Joi.string(),

})

export const validateTodo = (todo) => JoiSchema.validate(todo)

export default TodoModel

File: Todo app/todo/lib/models/todo.dart
class Todo {
  String? id, title, description, due, userid;
  // bool? complete;

  Todo({
    this.id,
    // this.complete,
    this.description,
    this.due,
    this.title,
    this.userid,
  });

  factory Todo.fromJson(var json) {
    return Todo(
      userid: json['userid'],
      title: json['title'],
      // complete: json['complete'],
      description: json['description'],
      due: json['due'],
      id: json['_id'],
    );
  }

  Map<String, dynamic> toJson() {
    Map<String, dynamic> json = <String, dynamic>{};
    json['userid'] = userid ?? '';
    json['title'] = title ?? '';
    // json['complete'] = complete ?? false;
    json['description'] = description ?? '';
    json['due'] = due ?? '';
    return json;
  }
}
